server/mailrepository/mailrepository-api/src/test/java/org/apache/james/mailrepository/MailRepositoryTest.java

```diff
@@ -0,0 +1,125 @@
+package org.apache.james.mailrepository;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.StringJoiner;
+
+import javax.mail.MessagingException;
+
+import org.apache.commons.lang.RandomStringUtils;
+import org.apache.james.core.MailAddress;
+import org.apache.james.mailrepository.api.MailRepository;
+import org.apache.james.mime4j.io.InputStreams;
+import org.apache.james.server.core.MailImpl;
+import org.apache.mailet.Mail;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import com.google.common.collect.ImmutableList;
+
+public interface MailRepositoryTest {
```

Comment:
Is it a test? Can I run it?

I like the standard Junit denomination: "**MailRepositoryContract**". Then implementation can provide test classes that implements these contracts, like **JDBCMailRepositoryTest**

Reply:
As discussed during meeting, I'll go for Contract

Reply:
changed